docs/user-interface/handlers/customize.md

```diff
@@ -1,122 +1,316 @@
+        {
+            ChangedHandler(sender, e);
+        }
+
+        void OnHandlerChanging(object sender, HandlerChangingEventArgs e)
+        {
+            ChangingHandler(sender, e);
+        }
     }
 }
 ```
 
-You can then customize the `EntryHandler` to perform the desired modification to `MyEntry` instances:
+In this example, the two event handlers call partial methods named `ChangedHandler` and `ChangingHandler`, whose signatures are defined in the cross-platform partial class. The partial method implementations are then defined in the platform-specific partial class, that's located in the **Platforms** > **Android** folder of the project:
 
 ```csharp
+using AndroidX.AppCompat.Widget;
 using Microsoft.Maui;
 using Microsoft.Maui.Controls;
 using Microsoft.Maui.Graphics;
+using System;
 
-namespace MauiApp1
+namespace HandlersDemos.Views
 {
-    public partial class App : Application
+    public partial class CustomizeEntryPage : ContentPage
     {
-        public App()
+        partial void ChangedHandler(object sender, EventArgs e)
         {
-            InitializeComponent();
+            ((sender as Entry).Handler.NativeView as Android.Views.View).FocusChange += OnFocusChange;
+        }
 
-            Microsoft.Maui.Handlers.EntryHandler.EntryMapper[nameof(IView.Background)] = (handler, view) =>
+        partial void ChangingHandler(object sender, HandlerChangingEventArgs e)
+        {
+            if (e.OldHandler != null)
             {
-                if (view is MyEntry)
-                {
-#if __ANDROID__
-                    handler.NativeView.SetBackgroundColor(Colors.Red.ToNative());
-#elif __IOS__
-                  handler.NativeView.BackgroundColor = Colors.Red.ToNative();
-                  handler.NativeView.BorderStyle = UIKit.UITextBorderStyle.Line;
-#elif WINDOWS
-                  handler.NativeView.Background = Colors.Red.ToNative();
-#endif
-                }
-            };
+                (e.OldHandler.NativeView as Android.Views.View).FocusChange -= OnFocusChange;
+            }
+        }
+
+        void OnFocusChange(object sender, EventArgs e)
+        {
+            var nativeView = sender as AppCompatEditText;
+
+            if (nativeView.IsFocused)
+                nativeView.SetBackgroundColor(Colors.LightPink.ToNative());
+            else
+                nativeView.SetBackgroundColor(Colors.White.ToNative());
         }
     }
 }
 ```
 
-Any `MyEntry` instances will then be customized as per the handler modification.
+The advantage of this approach is that compiler preprocessing directives aren't required, and that the partial methods don't have to be implemented on each platform. Instead, the compiler removes references to the method signature on any platforms that don't provide an implementation. For information about partial methods, see [Partial methods](/dotnet/csharp/programming-guide/classes-and-structs/partial-classes-and-methods#partial-methods).
```

Comment:
> the compiler removes references to the method signature on any platforms that don't provide an implementation.

Is this something special to MAUI? I've never heard of this before. So when compiling for iOS the compiler would remove the `ChangedHandler(sender, e);` call instead of erroring out that a `ChangedHandler` method doesn't exist?

Reply:
I've re-written this, but yes, there's no error about `ChangedHandler` not existing if it's not implemented on a platform.

The .NET partial method doc says (about partial methods) "If the developer does not supply an implementation, the compiler removes the signature at compile time.".